bluepysnap/network.py

```diff
@@ -177,9 +170,9 @@ def get(self, group=None, properties=None):
                 # However, it's a bit more performant than converting the Series to numpy arrays.
                 pop_df = pd.DataFrame({prop: pop.get(pop_ids, prop) for prop in pop_properties})
                 pop_df.index = global_pop_ids.index
-                dataframes.append(pop_df)
-        res = pd.concat(dataframes)
-        assert res.index.is_monotonic_increasing, "The index should be already sorted"
+
+                # Sort the columns in the given order
+                res.append((name, pop_df[[p for p in properties if p in pop_properties]]))
         return res
```

Comment:
Also, should we return a list, tuple or a generator?

Reply:
I prefer a generator, so the memory usage can be kept lower when not all the dataframes are needed at the same time.